docs/references/ayx-plugin-cli.md

```diff
@@ -0,0 +1,487 @@
+tools are installed under the **Python SDK Examples** category.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+-   Default: Python SDK Examples
+
+#### \--description
+
+The package description.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--author
+
+The plugin author.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--company
+
+The company creating the plugin.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--backend-language
+
+The programming language that a plugin will use. At present, the CLI
+only supports Python plugins.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: python
+
+## create-ayx-plugin
+
+This command generates the boilerplate code, configs, and UI for a new
+Alteryx plugin in the workspace.
+
+Note: If this command fails midway through execution, it leaves
+half-finished files and directories in the workspace. In order to delete
+these, follow these steps:
+
+-   Remove config/\<Plugin\> directory.
+-   Remove backend/ayx_plugins/\<plugin\>.py.
+-   Force-remove ui/\<Plugin\>.
+-   Remove the import statement from \_\_init\_\_.py.
+-   Remove the tool from ayx_workspace.json's "tools" key.
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
+
+#### \--tool-name
+
+This parameter becomes the name of the tool. Spaces are allowed but will
+be cleaned into underscores and Pascal case. For example, for a Python
+plugin named "A Test Tool", the filename is `a_test_tool.py`, and the
+`class/ui/config` name is `ATestTool`.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: Any alphanumeric string that doesn't start with
+    a number and doesn't shadow reserved names.
+
+#### \--tool-type
+
+This parameter is used to determine the scaffold that the tool should
+generate for a plugin.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: input, output, multiple-inputs,
+    multiple-outputs, single-input-single-output
+-   Default: single-input-single-output
+
+#### \--description
+
+The plugin description.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--version
+
+The plugin version.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+## create-yxi
+
+This command packages a workspace into a YXI that can be installed into
+Designer. The resulting YXI is located under `build/yxi/`.
+
+### Parameters
+
+This command takes no parameters.
+
+## install-yxi
+
+This command takes a YXI (any CLI-built or legacy SDK YXI) and installs
+it into Designer.
+
+Note: This command can be run from anywhere, not just Alteryx Workspace
+directories.
```

Comment:
TW review:

Note: You can run this command from anywhere, not just Alteryx Workspace
directories.

Reply:
changed